source/NoteRefFilter.cpp

```diff
@@ -0,0 +1,43 @@
+// Header file
+#include "NoteRefFilter.h"
+
+// System libraries
+#include <NodeInfo.h>
+
+// Filter hook method
+bool NoteRefFilter :: Filter(const entry_ref *ref,
+			BNode *_node,
+			struct stat_beos *stat,
+			const char *_mimeType) {
+
```

Comment:
Because this is a new file you would add, please follow the [coding guidelines](https://www.haiku-os.org/development/coding-guidelines/):
```
bool
NoteRefFilter::Filter(const entry_ref* ref, BNode* _node,
	struct stat_beos* stat, const char* _mimeType)
{
```